ceph/rados/serviceability_workflows.py

```diff
@@ -494,20 +514,32 @@ def drain_host(self, host_obj, zap=True, keep_conf=False):
                             "drain process for OSD %s is still going on"
                             % entry["osd_id"]
                         )
-                        if end_time < datetime.datetime.now():
-                            log.error(
-                                "Could not drain OSD %s within timeout"
-                                % entry["osd_id"]
-                            )
-                            drain_success = False
-                        log.info("Sleeping for 120 seconds and checking again....")
-                        time.sleep(120)
-                    log.info("Drain operation completed for OSD %s" % entry["osd_id"])
-                log.info("Drain operations completed on host : " + host_obj.hostname)
-                break
+                log.info("Sleeping for 20 seconds and checking again....")
+                time.sleep(20)
             except json.JSONDecodeError:
                 log.info("Drain operations completed on host : " + host_obj.hostname)
-                break
+                # Add checks for 8.1 OSD process not removed.
```

Comment:
Add the bug raised for 8.x in comments. 

With the Next PR, add WA for the 8.x bug.